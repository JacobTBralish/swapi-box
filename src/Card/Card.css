.card {
  width: 200px;
  height: 200px;
  border: 2px solid black;
}

.card-line {
  display: block;
}