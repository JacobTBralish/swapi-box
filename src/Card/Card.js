import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

export const Card = (props) => {
  const propsArray = Object.entries(props);
  const renderProps = propsArray.slice(0, propsArray.length - 2);
  if (props.style) {
    renderProps.pop();
  }

  const linesToRender = renderProps.map( entry => {
    return <li 
      className='card-line title' 
      key={entry[0]}> { entry[0] } : 
      <span className='content'> { entry[1] } </span> 
    </li>;
  });

  return (
    <div className={`card ${props.style}`} 
      onClick={props.handleClick} 
      id={props.id}>
      {linesToRender}
    </div>
  );
};

Card.propTypes = {
  id: PropTypes.string.isRequired,
  style: PropTypes.string,
  handleClick: PropTypes.func.isRequired
};