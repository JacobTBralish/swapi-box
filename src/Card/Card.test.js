import React from 'react';
import ReactDOM from 'react-dom';
import { Card } from './Card';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Card Component', () => {
  test('that it renders without crashing', () => {
    const renderedComp = shallow(<Card />)

    expect(renderedComp).toMatchSnapshot()
  })

  test('that it should receive and render props', () => {
    const renderedComp = shallow(<Card name="Luke Skywalker" />)

    expect(renderedComp).toMatchSnapshot()
  })
})