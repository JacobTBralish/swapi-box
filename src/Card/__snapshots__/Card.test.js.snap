// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component that it renders without crashing 1`] = `
<div
  className="card undefined"
/>
`;

exports[`Card Component that it should receive and render props 1`] = `
<div
  className="card undefined"
/>
`;
